smoke/esm/src/index.test.ts

```diff
@@ -1,60 +1,115 @@
+  label.style.backgroundColor = "white";
+  label.style.padding = "10px";
+  label.style.border = "1px solid black";
+  label.style.fontFamily = "monospace";
+  document.body.appendChild(label);
+
+  // Wait a moment for the DOM to stabilize
+  await new Promise((resolve) => setTimeout(resolve, 1000));
 
-it("esm.sh", async () => {
   const script = document.createElement("script");
   // eslint-disable-next-line no-console
-  console.log("FP_VERSION", (window as unknown as { FP_VERSION: string }).FP_VERSION);
-  // eslint-disable-next-line no-console
-  console.log("FP_DEBUG", (window as unknown as { FP_DEBUG: string }).FP_DEBUG);
-  // eslint-disable-next-line no-console
-  console.log("FP_STACK", (window as unknown as { FP_STACK: string }).FP_STACK);
-  // const res = await fetch(`http://localhost:4874/@fireproof/core@${window.FP_VERSION}?no-dts`);
-  // // console.log("window-res", await res.text());
-  // const { fireproof } = await import(/* @vite-ignore */ `http://localhost:4874/@fireproof/core@${window.FP_VERSION}?no-dts`);
-  // // console.log("window-imp", fireproof);
-
+  console.log("Running ESM test");
   script.textContent = `
-//console.log("pre-window-js", window.FP_VERSION)
-import { fireproof } from 'http://localhost:4874/@fireproof/core@${window.FP_VERSION}?no-dts'
+    // Create a function to update the test status
+    function updateTestStatus(status, message) {
+      const label = document.getElementById('test-result');
+      if (label) {
+        label.textContent = status + ': ' + message;
+        label.setAttribute('data-status', status);
+      }
+    }
 
-console.log("window-js", window.FP_VERSION)
-function invariant(cond, message) {
-  if (!cond) {
-    throw new Error(message)
-  }
-}
-async function action(label, run) {
-  const db = fireproof("esm-test");
-  const ok = await db.put({ sort: Math.random(), test: "esm-success" });
-
-  const beforeAll = await db.allDocs()
-  await db.put({ foo: 1 })
-  const afterAll = await db.allDocs()
-
-  invariant(
-     afterAll.rows.length == beforeAll.rows.length + 1,
-     "all docs wrong count",
-  )
-
-  const res = await db.get(ok.id)
-  label.innerHTML = [run,res.test].join(' - ')
-  await db.close()
-}
-
-async function main() {
-  const label = document.querySelector('label')
-  for (let i = 0; i < 10; i++) {
-    await action(label, i)
-  }
-  label.setAttribute("data-ready", "");
-}
-main().catch(console.error)
-`;
+    // Set initial status
+    updateTestStatus('RUNNING', 'Importing Fireproof...');
+
+    // Handle errors globally
+    window.addEventListener('error', function(event) {
+      console.error('Global error:', event.error);
+      updateTestStatus('ERROR', 'Global error: ' + event.error.message);
+    });
+
+    // Run the test in an async IIFE
+    (async function runTest() {
+      try {
+        updateTestStatus('RUNNING', 'Importing Fireproof module...');
+        const fireproof = await import('https://esm.sh/@fireproof/core@latest');
```

Comment:
need the explict version!!!!!

Reply:
like
import { fireproof } from 'http://localhost:4874/@fireproof/core@${window.FP_VERSION}?no-dts'